src/Sylius/Bundle/CoreBundle/DiscountApplicationCriteria/ExclusiveCriteria.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\DiscountApplicationCriteria;
+
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Core\Model\ChannelPricingInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionActionInterface;
+
+final class ExclusiveCriteria implements DiscountApplicationCriteriaInterface
+{
+    public function isApplicable(CatalogPromotionInterface $catalogPromotion, CatalogPromotionActionInterface $action, ChannelPricingInterface $channelPricing): bool
```

Comment:
Do you need all of these arguments? in both criteria you use only `channelPricing`

Reply:
To make it more generic i decided to pass all arguments to make it easy to add new Criteria. I can remove it, but i think this way we allow everyone to make custom criteria with ease.

Reply:
I've shared my opinion [here](https://github.com/Sylius/Sylius/pull/13499#discussion_r786624429) 🖖